### DISCUSSION TITLE
GameCube 1Player Adapter

### DISCUSSION THREAD

**Discussion by alastairbdoggett** [darthcloud/BlueRetro#63] [Show and tell]
Finished my first BlueRetro build! Proof that even a complete wiring/electronics novice can build the project :)

In case anyone else is in the position of being a little intimidated by the hardware, I thought I would document the process briefly here:

Equipment:
- HiLetgo ESP-WROOM-32 ESP32 ESP-32S Development Board
- Male to Female Dupont Wires
- Ssgamer 6ft Wii/GameCube extension cables
- GKEEMARS SPL-1 (inline connectors)
- Digital Multimeter

Steps:
1. Even though i'm not a Windows user, the process of flashing the firmware seemed much more straight forward from a Windows box so I went down that route. In addition to the instructions provided in the wiki, I manually installed the [CP210x driver](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/establish-serial-connection.html). I only had a single COM port to choose from but in `Device Manager > Ports (COM & LPT)` I saw something like `CP210x USB to UART Bridge (COM3)` which I made sure matched with what was available in the `ESP32 DOWNLOAD TOOL`. Since I was only interested in GC gaming, I used this guy `BlueRetro_gc_spiffs.bin` instead of `BlueRetro.bin`.
2. To be extra safe, I checked that I could pair the flashed esp32 board, I used the [Web Bluetooth](https://blueretro.io) as per the instructions. Everything appeared to be ok 🤞 
3.  I then performed continuity tests with the multimeter to figure out which wire was connected to which component in the connector port (I asked a [question about interpreting orientation](https://github.com/darthcloud/BlueRetro/discussions/61#discussion-3527956) in case it's not clear). I used the inline connector to connect the exposed wires from the controller cable to the Dupont wires (in my case, the controller cables were very thin so it was a little fiddly 😬). To check the connections were good, I performed the continuity checks once more, this time with one terminal being the Dupont wire.
4. Most of the work was done at this point. I then followed [this advice](https://github.com/darthcloud/BlueRetro/discussions/46#discussioncomment-928618) to make sure (again) that everything was as it should be (I was a little worried about [possibly bricking the console](https://www.reddit.com/r/Gamecube/comments/it4gq5/5v_short_on_power_board/) and am a complete beginner to this sort of stuff, hence the extra sanity checks!)
5. Finally, I booted up the console and using a PS4 controller paired without a hitch the first time. To make sure, I power cycled the console and the same controller paired again without a hitch.

Really enjoyed this project and massive thanks to @darthcloud for this awesome work and to @vaderag who assisted me with some tips and tricks that they had learned ☺️

**Comment by alastairbdoggett** [comment]
![IMG_7994](https://user-images.githubusercontent.com/17734670/129995537-9d10d58a-b38a-4888-ad5c-e910616af9e0.jpg)

**Reply by darthcloud** [reply]
Well done! Enjoy :)

**Reply by alastairbdoggett** [reply]
![blueretro_gcn](https://user-images.githubusercontent.com/17734670/131003501-51d8c563-d41f-485f-ac9e-505953875bd4.jpg)

upgraded to 4x ports! cables made by [@GH0ST0X0](https://twitter.com/GH0ST0X0) :)

**Reply by alastairbdoggett** [reply]
just need something to house the board + wires now

**Reply by vaderag** [reply]
Nice cables!
